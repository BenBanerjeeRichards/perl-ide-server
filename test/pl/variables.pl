use warnings;
use strict;

$hello = 2;
$_Hello = 3;
%hello = 3;
@_83Hel34 = 5;
$H3ello = 54;
$::::'foo;
$foo::'bar;
$Hello::world::thing;
$World::Hello'thing;

$1 = 2;
$0 = 3;
$20384 = 23;

$!;
$?;
$^;

$^A;
$^B;
$^?;
$^^;

${^HE34LLO};
${^Hello};
${^_Hell2o};

# Single dereferences
$$hello;

# Multiple dereferences
$$$hello;
%$$hello;


# These are not valid and should not be interpreted as variables
${Hello};
$::'::foo;
$foo'::bar;
${^Hello;
