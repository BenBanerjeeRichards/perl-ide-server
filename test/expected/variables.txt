Use
Name(warnings)
Semicolon
Use
Name(strict)
Semicolon
ScalarVariable($hello)
Assignment
NumericLiteral(2)
Semicolon
ScalarVariable($_Hello)
Assignment
NumericLiteral(3)
Semicolon
HashVariable(%hello)
Assignment
NumericLiteral(3)
Semicolon
ArrayVariable(@_83Hel34)
Assignment
NumericLiteral(5)
Semicolon
ScalarVariable($H3ello)
Assignment
NumericLiteral(54)
Semicolon
ScalarVariable($::::'foo)
Semicolon
ScalarVariable($foo::'bar)
Semicolon
ScalarVariable($Hello::world::thing)
Semicolon
ScalarVariable($World::Hello'thing)
Semicolon
ScalarVariable($1)
Assignment
NumericLiteral(2)
Semicolon
ScalarVariable($0)
Assignment
NumericLiteral(3)
Semicolon
ScalarVariable($20384)
Assignment
NumericLiteral(23)
Semicolon
ScalarVariable($!)
Semicolon
ScalarVariable($?)
Semicolon
ScalarVariable($^)
Semicolon
ScalarVariable($^A)
Semicolon
ScalarVariable($^B)
Semicolon
ScalarVariable($^?)
Semicolon
ScalarVariable($^^)
Semicolon
ScalarVariable(${^HE34LLO})
Semicolon
ScalarVariable(${^Hello})
Semicolon
ScalarVariable(${^_Hell2o})
Semicolon
Comment(# Single dereferences)
Deref($)
ScalarVariable($hello)
Semicolon
Comment(# Multiple dereferences)
Deref($)
Deref($)
ScalarVariable($hello)
Semicolon
Deref(%)
Deref($)
ScalarVariable($hello)
Semicolon
Comment(# Use $# for array size)
ScalarVariable($#array)
Semicolon
ScalarVariable($#$array_ref)
Semicolon
Comment(# These are not valid and should not be interpreted as variables)
Name($)
LBracket
Name(Hello)
RBracket
Semicolon
ScalarVariable($:)
Operator(:)
String('::foo;\n$foo')
Operator(:)
Operator(:)
Name(bar)
Semicolon
Name($)
LBracket
Operator(^)
Name(Hello)
Semicolon
EndOfInput
